addons/account/models/account_tax.py

```diff
@@ -1262,6 +1262,29 @@ class AccountTaxRepartitionLine(models.Model):
         help="The order in which distribution lines are displayed and matched. For refunds to work properly, invoice distribution lines should be arranged in the same order as the credit note distribution lines they correspond to.")
     use_in_tax_closing = fields.Boolean(string="Tax Closing Entry", default=True)
 
+    is_multi_vat_company = fields.Boolean(string="Is its company a multi vat company", related="company_id.is_multi_vat_company")
+    tag_ids_domain = fields.Binary(string="Dynamic domain used for the tag that can be set on tax", compute="_compute_tag_ids_domain")
+
+    @api.depends("tax_id", "is_multi_vat_company")
+    def _compute_tag_ids_domain(self):
+        company_to_foreign_vat_country = {}
+        # TODO: discuss with reviewer
+        #  If I make the read_group everytime, I don't need the is_multi_vat_company field on this model
+        if any(self.mapped("is_multi_vat_company")):
+            company_to_foreign_vat_country = {
+                val["company_id"][0]: val["country_ids"]
+                for val in self.env["account.fiscal.position"].read_group(
+                    domain=[("company_id", "in", self.company_id.ids), ("foreign_vat", "!=", False)],
+                    fields=["country_ids:array_agg(country_id)"],
+                    groupby="company_id",
+                )
+            }
```

Comment:
It's only used in the view and thus it should be a problem to `read_group` every time